File: C#/Longest Palindromic Substring/Longest Palindromic Substring.cs
public class Solution {
    public string LongestPalindrome(string s) {
        if(s.Length<=1) return s;
        string ret=s.Substring(0,1);
        int mx=1;
        bool[,] dp=new bool[s.Length,s.Length];
        for(int i=1;i<s.Length;i++){
            for(int j=0;j<i;j++){
                if(i-j<3 && s[i]==s[j])
                    dp[i,j]=true;
                else
                    dp[i,j]=dp[i-1,j+1] && (s[i]==s[j]);
                if(dp[i,j]){
                    int m=i-j+1;
                    if(m>mx){
                        mx=m;
                        ret=s.Substring(j,i-j+1);
                    }
                }
            }
        }
        return ret;
    }
}

File: Python/Longest Palindromic Substring/Longest Palindromic Substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s=='':
            return ''
        n=len(s)
        ret,mx=s[0],1
        dp=[[False]*n for _ in range(n)]
        for j in range(1,n):
            for i in range(0,j):
                if j-i<3 and s[i]==s[j]:
                    dp[i][j]=True
                else:
                    dp[i][j]=dp[i+1][j-1] and (s[i]==s[j])
                if dp[i][j]:
                    m=j-i+1
                    if m>mx:
                        ret=s[i:j+1]
                        mx=m
        return ret
                    